### DISCUSSION TITLE
Version 1.10.0 - KiCad integration

### DISCUSSION THREAD

**Discussion by jbtronics** [Part-DB/Part-DB-server#452] [Announcements]
# Part-DB 1.10.0

**If you like Part-DB, please consider donating to help further development. See the sponsor button on the [homepage](https://github.com/Part-DB/Part-DB-server).**

## KiCad integration

Part-DB now supports the experimental HTTP library feature in the KiCad nightly builds (KiCad 7.99). It allows to define metadata like the KiCad footprints and symbols on Part-DB parts and use Part-DB as libray inside KiCad.
See [here](https://docs.part-db.de/usage/eda_integration.html) for more information.

If you want to try it out, install the latest KiCad nightly build and create the following file with the name `partdb_demo.kicad_httplib` with the following content and add it as library inside KiCad. This will load parts from the [Part-DB demo instance](https://demo.part-db.de/en/):

**For performance reasons this was disabled on the demo instance. Please try it on your own Part-DB instance.**

```
{
    "meta": {
        "version": 1.0
    },
    "name": "Part-DB demo",
    "description": "A KiCad library sourced from the Part-DB demo instance",
    "source": {
        "type": "REST_API",
        "api_version": "v1",
        "root_url": "https://demo.part-db.de/en/kicad-api/",
        "token": "tcp_d985d460a387a55bd5076766f0ccf71094af7a93cc4ad38879c44c8370e97097"
    }
}

```

 ## Full changelog below:

_Database schema changed: You need to run `php bin/console doctrine:migrations:migrate` after upgrade!_

## New features
* Added KiCad integration, which allows to use Part-DB as a datasource inside KiCad. See [docs](https://docs.part-db.de/usage/eda_integration.html) for more info.

## Bugfixes
* Show a flash warning, when trying to withdraw/add an amount of 0 to parts instead of failing (issue #448)

## Miscellaneous
* Updated translations
* Updated dependencies
* Updated symfony to 6.4
* Fixed some deprecations

**Full Changelog**: https://github.com/Part-DB/Part-DB-server/compare/v1.9.1...v1.10.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Part-DB/Part-DB-server/releases/tag/v1.10.0'>Version 1.10.0</a>.</em>

**Comment by Pyromane** [comment]
Thank you for checking if the correct console user is used in update 1.10.1!

**Comment by rdalen** [comment]
I expected to be able to define the reference-prefix at category-level in Part-DB, but i notice that only a reference prefix on part-level shows up in KiCad. Otherwise every part gets the prefix U in KiCad. Is this a Part-DB issue or a KiCad issue?

**Reply by jbtronics** [reply]
This was actually a bug in Part-DB and the reference prefix setting on categories was not used.
This is fixed in Part-DB 1.11.3

**Reply by rdalen** [reply]
Thnx! - It works when i define the reference prefix in the same category as where i define the symbol - in my example in de category "common diodes". When i define the reference prefix one level higher - in my example in the category "Diodes" so that the reference prefix applies to all types of diodes - then it doesn't work.
![Catagories](https://github.com/Part-DB/Part-DB-server/assets/11911108/e0a4680b-c60d-40a4-8254-3e28275ae987)

**Reply by jbtronics** [reply]
Yes thats currently the intended behavior. The reference prefix does not inherit to sub categories.

I would expect that it should not be too much work to set this on all sub categories.
